### DISCUSSION TITLE
own snippets?

### DISCUSSION THREAD

**Discussion by msccto** [alefragnani/vscode-language-pascal#121] [Q&A] [answered]
Hi,
I just found this really good looking extension and started playing around. 
One question: is there a way to add own snippets or modify the existing ones?

thx
Matthias

**Comment by alefragnani** [comment, marked as answer]
Hi @msccto ,

First of all, thank you. I’m glad to know you like the extension.

About the snippets, most of those provided by the extension were inspired by the ones you find in Delphi, because it was my main development language/IDE. And the idea to provide it in the extension was to make easier for me (and any other Delphi developer) to migrate/interchange to VS Code. If you found some issue in any of those snippets, or if you have snippets that makes sense for any Pascal developer, feel free to open a PR and contribute to the extension.

Otherwise, you have two alternatives on how to deal with Snippets in VS Code:

* If you have snippets that only makes sense for a set of developers (maybe because it is related to some library/component you use) but you would like to share with other users, you could create a new Snippets Extension and publish it yourself via [Language Extensions / Snippets Guide](https://code.visualstudio.com/api/language-extensions/snippet-guide)
* If you have snippets that only makes sense for you, you could add it to your own VS Code instance, via [User Defined Snippets](https://code.visualstudio.com/docs/editor/userdefinedsnippets)

Hope this helps
